sdk/typescript/src/builder/TransactionBlock.ts

```diff
@@ -310,56 +317,116 @@ export class TransactionBlock {
 		return this.object(Inputs.SharedObjectRef(...args));
 	}
 
-	/**
-	 * Add a new non-object input to the transaction.
-	 */
-	pure(
-		/**
-		 * The pure value that will be used as the input value. If this is a Uint8Array, then the value
-		 * is assumed to be raw bytes, and will be used directly.
-		 */
-		value: unknown,
-		/**
-		 * The BCS type to serialize the value into. If not provided, the type will automatically be determined
-		 * based on how the input is used.
-		 */
-		type?: string,
-	) {
+	pure = createPure((value, type) => {
+		if (isSerializedBcs(value)) {
+			return this.#input('pure', {
+				Pure: Array.from(value.toBytes()),
+			});
+		}
+
 		// TODO: we can also do some deduplication here
 		return this.#input(
 			'pure',
 			value instanceof Uint8Array ? Inputs.Pure(value) : type ? Inputs.Pure(value, type) : value,
 		);
-	}
+	});
 
 	/** Add a transaction to the transaction block. */
 	add(transaction: TransactionType) {
 		const index = this.#blockData.transactions.push(transaction);
 		return createTransactionResult(index - 1);
 	}
 
+	#normalizeTransactionArgument(
+		arg: TransactionArgument | SerializedBcs<any>,
+	): TransactionArgument {
+		if (isSerializedBcs(arg)) {
+			return this.pure(arg);
+		}
+
+		return arg as TransactionArgument;
+	}
+
 	// Method shorthands:
 
-	splitCoins(...args: Parameters<(typeof Transactions)['SplitCoins']>) {
-		return this.add(Transactions.SplitCoins(...args));
+	splitCoins(
+		coin: TransactionObjectArgument,
```

Comment:
not sure if this is a good idea, basically this is omitting the ability to pass pure arguments in places we know should be objects

Reply:
it did require updating a couple places in the kiosk sdk to use `TransactionObjectArgument` instead of `TransactionArgument`

Reply:
I think this is probably good, I'd rather more narrow types here.